internal/procedures/server/start.go

```diff
@@ -46,10 +47,13 @@ func Start() (err error) {
 		go startCourse(course)
 	}
 
-	// Cleanup any temp dirs.
-	defer util.RemoveRecordedTempDirs()
+	// Keep temp dirs for the CMD testing infrastructure to access after this function ends.
+	if !config.UNIT_TESTING_MODE.Get() {
+		// Cleanup any temp dirs.
+		defer util.RemoveRecordedTempDirs()
+	}
```

Comment:
Is this necessary?
You call pserver.RunSever(), not pserver.Start() when you start with `common.CMD_TEST_SERVER`.

If it is necessary, I think there is a better way.
You don't want to directly check `config.UNIT_TESTING_MODE` if you don't have to,
checking it represents a special case for testing.
But here, we can just pass a `cleanup` boolean argument which makes sense to have in any context.

Reply:
It is necessary for the one test where the cmd starts its own server. Should the comment better reflect this one case or is it fine to keep it saying it's for the CMD testing infrastructure because that one case also uses the infrastructure?